pkg/sparrow/run.go

```diff
@@ -116,8 +113,31 @@ func (s *Sparrow) Run(ctx context.Context) error {
 			s.cErr <- s.tarMan.Reconcile(ctx)
 		}
 	}()
+
 	go func() {
-		s.cErr <- s.api(ctx)
+		routes := []api.Route{
+			{
+				Path: "/openapi", Method: http.MethodGet,
+				Handler: s.handleOpenAPI,
+			},
+			{
+				Path: fmt.Sprintf("/v1/metrics/{%s}", urlParamCheckName), Method: http.MethodGet,
+				Handler: s.handleCheckMetrics,
+			},
+			{
+				Path: "/metrics", Method: "Handle",
+				Handler: promhttp.HandlerFor(
+					s.metrics.GetRegistry(),
+					promhttp.HandlerOpts{Registry: s.metrics.GetRegistry()},
+				).ServeHTTP,
+			},
+		}
+		err := s.api.RegisterRoutes(ctx, routes...)
+		if err != nil {
+			log.Error("Error while registering routes", "error", err)
+			s.cErr <- err
+		}
+		s.cErr <- s.api.Run(ctx)
```

Comment:
For now I've just moved the API startup from the `Run` method to another method of the sparrow because the `/openapi` handler currently needs the sparrows check map to properly generate the check oapi specs during runtime and it'd feel awkward to have multiple places for the handlers.

This will also make the refactoring of the sparrow's database responsibility a bit tricky so therefore it should be done with/after #98